### PR TITLE
Add more information on config files in GMF

### PR DESCRIPTION
**Pull request by yjacolin**
linkedin_manadger1
linkedin_manadger1

And:

linkedin_manadger1

### CODE REVIEW

File: doc/integrator/create_application.rst [hunk 1]
```diff
@@ -297,6 +370,10 @@ To be consistent with c2cgeoportal terminology we will use the words `project`
 to refer to the whole project and `instance` for a dedicated configuration of
 the project.
 
+Multi-instance project use the concepts describe above:
+
+.. image:: ../_static/doc_hierarchie_multiproject.png
```
Comment (sbrunner) [inline doc/integrator/create_application.rst]: It this missing?

File: doc/integrator/create_application.rst [hunk 2]
```diff
@@ -226,6 +226,80 @@ Commit and push on the main repository:
...
+
+Hierarchy and extending your configuration files
+................................................
+
+The configuration files (Makefile and vars) have a hierarchy between them.
+These files extend other files. A Make file extends another Make file
+and similarly a vars file extends another vars file. This extension is visible
+in Makefile files::
+
+    extends: CONST_vars.yaml
```
Comment (sbrunner) [inline doc/integrator/create_application.rst]: ???, it's the vars version...

File: doc/integrator/create_application.rst [hunk 3]
```diff
@@ -226,6 +226,80 @@ Commit and push on the main repository:
...
+and vars files::
+
+    extends: CONST_vars.yaml
+
+CONST files are files that should not be changed because they are replaced
+during application updates, so your changes will be systematically lost. You can
+extend these files as many times as you like, although it is not recommended to
+exceed 3-4 levels for readability and simplicity.
+
+.. image:: ../_static/doc_hierarchie.png
```
Comment (sbrunner) [inline doc/integrator/create_application.rst]: missing image

File: doc/integrator/create_application.rst [hunk 4]
```diff
@@ -226,6 +226,80 @@ Commit and push on the main repository:
...
+
+Whenever possible, it is strongly advised not to extend the
+`vars_myProject.yaml` file, and we recommend that you use dynamic variables as
```
Comment (sbrunner) [inline doc/integrator/create_application.rst]: we use : `vars_<project>.yaml`

File: doc/integrator/create_application.rst [hunk 5]
```diff
@@ -226,6 +226,80 @@ Commit and push on the main repository:
...
+
+* Configuring really different environments.
+* Configuration of a multi-project (see below for this specific use case).
+
+Use of dynamic variable
+.......................
+
+Variables used in the application configuration files (files `vars_xxx.yaml`)
+can be made dynamic by means of environment variable. In the main file
+`vars_myProject.yaml`, added the `interpreted` block at the bottom of the
```
Comment (sbrunner) [inline doc/integrator/create_application.rst]: also

File: doc/integrator/create_application.rst [hunk 6]
```diff
@@ -226,6 +226,80 @@ Commit and push on the main repository:
...
+interpreted parameters section::
+
+    extends: CONST_vars.yaml
+
+    vars:
+        host: HOST
+    [..]
+    interpreted:
+        environment:
+            - log_level
```
Comment (sbrunner) [inline doc/integrator/create_application.rst]: should be ... or removed ...

File: doc/integrator/create_application.rst [hunk 7]
```diff
@@ -226,6 +226,80 @@ Commit and push on the main repository:
...
+            - host
+
+In the `MyProject.mk` file, add parameters you want to change and the
```
Comment (sbrunner) [inline doc/integrator/create_application.rst]: also

File: doc/integrator/create_application.rst [hunk 8]
```diff
@@ -226,6 +226,80 @@ Commit and push on the main repository:
...
+`ENVIRONMENT_VARS` parameter::
```
Comment (sbrunner) [inline doc/integrator/create_application.rst]: We don't use `ENVIRONMENT_VARS` anymore, use `export` in place :-) (as you do bellow :-) )

File: doc/integrator/create_application.rst [hunk 9]
```diff
@@ -226,6 +226,80 @@ Commit and push on the main repository:
...
+    export HOST = domaine.different.com
+    export LOG_LEVEL
+
+In the Make files that extend this main file, you only need to define the
+environment variables::
+
+   export HOST = prod.different.com
+
+
+
```
Comment (sbrunner) [inline doc/integrator/create_application.rst]: 2 end line

File: doc/integrator/create_application.rst [hunk 10]
```diff
@@ -287,7 +361,6 @@ Then follow the sections in the install application guide:
     database creation as described in :ref:`integrator_install_application`,
     except the 'Get the application source tree' chapter.
 
-
```
Comment (sbrunner) [inline doc/integrator/create_application.rst]: 2 end line

File: doc/integrator/create_application.rst [hunk 11]
```diff
@@ -226,6 +226,80 @@ Commit and push on the main repository:
...
+            - log_level
+            - host
+
+In the `MyProject.mk` file, add parameters you want to change and the
+`ENVIRONMENT_VARS` parameter::
+
+    export HOST = domaine.different.com
+    export LOG_LEVEL
+
+In the Make files that extend this main file, you only need to define the
```
Comment (sbrunner) [inline doc/integrator/create_application.rst]: Make files => Makefiles

File: doc/integrator/create_application.rst [hunk 12]
```diff
@@ -226,6 +226,80 @@ Commit and push on the main repository:
...
+Use of dynamic variable
+.......................
+
+Variables used in the application configuration files (files `vars_xxx.yaml`)
+can be made dynamic by means of environment variable. In the main file
+`vars_myProject.yaml`, added the `interpreted` block at the bottom of the
+file.
+
+In this same file, you can change the value of a parameter by putting it in
+uppercase (example: host: HOST). This parameter must be listed in the
```
Comment (sbrunner) [inline doc/integrator/create_application.rst]: host: HOST in two back quote => ``host: HOST``

File: doc/integrator/create_application.rst [hunk 13]
```diff
@@ -226,6 +226,80 @@ Commit and push on the main repository:
...
+interpreted parameters section::
```
Comment (sbrunner) [inline doc/integrator/create_application.rst]: =>
interpreted parameters section:

.. code:: yaml

File: doc/integrator/create_application.rst [hunk 14]
```diff
@@ -226,6 +226,80 @@ Commit and push on the main repository:
...
+            - host
+
+In the `MyProject.mk` file, add parameters you want to change and the
+`ENVIRONMENT_VARS` parameter::
+
+    export HOST = domaine.different.com
+    export LOG_LEVEL
+
+In the Make files that extend this main file, you only need to define the
+environment variables::
```
Comment (sbrunner) [inline doc/integrator/create_application.rst]: =>
```
 ...variables:

.. code:: make
```

File: doc/integrator/create_application.rst (hunk 8 above)
Comment (sbrunner) [inline doc/integrator/create_application.rst]: =>
```
 ...parameter:

.. code:: make
```

File: doc/integrator/create_application.rst [hunk 15]
```diff
@@ -226,6 +226,80 @@ Commit and push on the main repository:
...
+................................................
+
+The configuration files (Makefile and vars) have a hierarchy between them.
+These files extend other files. A Make file extends another Make file
+and similarly a vars file extends another vars file. This extension is visible
+in Makefile files::
+
+    extends: CONST_vars.yaml
+
+and vars files::
```
Comment (sbrunner) [inline doc/integrator/create_application.rst]: =>
```
 ...files:

.. code:: make
```

File: doc/integrator/create_application.rst (hunk 4 above)
Comment (sbrunner) [inline doc/integrator/create_application.rst]: 2 back quote in restructured text

File: doc/integrator/create_application.rst (hunk 5 above)
Comment (sbrunner) [inline doc/integrator/create_application.rst]: 2 back quote in restructured text

File: doc/integrator/create_application.rst (hunk 7 above)
Comment (sbrunner) [inline doc/integrator/create_application.rst]: 2 back quote in restructured text

**Comment by sbrunner** [comment]
For me the first graph should be like this...
graph

**Comment by sbrunner** [comment]
It's good if you also commit the sources files of the diagrams

**Comment by sbrunner** [comment]
Thanks @yjacolin, many small comments but that's looks good :-)

**Comment by yjacolin** [comment]
@sbrunner I pushed my changes. Thanks for your comments :)

**Comment by rbovard** [comment]
Wouldn't be better to translate image legends?

File: doc/integrator/create_application.rst [hunk 16]
```diff
@@ -226,6 +226,88 @@ Commit and push on the main repository:
...
+* Makefile: These files are environment configuration files. Each environment will
+  have its configuration file (developer, preprod, prod)
+* vars_xxx.yaml: These files are application configuration files. Generally only
+  one file is needed to configure your application.
+
+Hierarchy and extending your configuration files
+................................................
+
+The configuration files (Makefile and vars) have a hierarchy between them.
+These files extend other files. A Make file extends another Make file
```
Comment (sbrunner) [inline doc/integrator/create_application.rst]: => Makefile

File: doc/integrator/create_application.rst [hunk 17]
```diff
@@ -226,6 +226,88 @@ Commit and push on the main repository:
...
+    vars:
+        host: HOST
+    ...
+    interpreted:
+        environment:
+            - log_level
+            - host
+
+In the ``<project>.mk`` file, add parameters you want to change and the
+``ENVIRONMENT_VARS`` parameter:
```
Comment (sbrunner) [inline doc/integrator/create_application.rst]: ... add parameters you want to change as export parameter:

**Comment by yjacolin** [comment]
@rbovard , @sbrunner done.

File: doc/integrator/create_application.rst [hunk 18]
```diff
@@ -226,6 +226,88 @@ Commit and push on the main repository:
...
+* Makefile: These files are environment configuration files. Each environment will
+  have its configuration file (developer, preprod, prod)
+* vars_xxx.yaml: These files are application configuration files. Generally only
+  one file is needed to configure your application.
+
+Hierarchy and extending your configuration files
+................................................
+
+The configuration files (Makefile and vars) have a hierarchy between them.
+These files extend other files. A Makefile extends another Make file
```
Comment (sbrunner) [inline doc/integrator/create_application.rst]: and the second one :-)

File: doc/integrator/create_application.rst [hunk 19]
```diff
@@ -226,6 +226,88 @@ Commit and push on the main repository:
...
+    vars:
+        host: HOST
+    ...
+    interpreted:
+        environment:
+            - log_level
+            - host
+
+In the ``<project>.mk`` file, add parameters you want to change as exported
+parameters and the ``ENVIRONMENT_VARS`` parameter:
```
Comment (sbrunner) [inline doc/integrator/create_application.rst]: I steel see the deprecated ENVIRONMENT_VARS...

Comment (yjacolin) [inline doc/integrator/create_application.rst]: oh, I misunderstood your last comment.

**Comment by yjacolin** [comment]
@sbrunner build failed, is it me or another commit? I guess this is not me :)

**Comment by sbrunner** [comment]
It's not you, be free to merge :-)

**Pull request merged**